src/System.Memory/src/System.Memory.csproj

```diff
@@ -9,6 +9,7 @@
     <DefineConstants Condition="'$(IsPartialFacadeAssembly)' != 'true'">$(DefineConstants);netstandard</DefineConstants>
     <DefineConstants Condition="'$(TargetGroup)'=='netcoreapp'">$(DefineConstants);netcoreapp</DefineConstants>
     <DefineConstants Condition="'$(TargetGroup)'=='netstandard1.1'">$(DefineConstants);netstandard11</DefineConstants>
+    <DefineConstants Condition="'$(ArchGroup)' == 'x86'">$(DefineConstants);BIT32</DefineConstants>
```

Comment:
If you want to do this as build setting, you need to do some work on the configurations - there is no separate configuration for System.Memory for x86 vs. amd64 today.

Reply:
The alternative is to do this as runtime check against pointer size - it is what System.Memory has been using so far.

Reply:
Is there another approach to toggle between the implementation based on the architecture that won't require having separate configurations for System.Memory?

Reply:
Ah I see. Something like this?

```C#
if (Unsafe.SizeOf<IntPtr>() == sizeof(long)) {}
else {}
```

Would this end up hurting perf?

Reply:
> Would this end up hurting perf?

5-15% regression depending on how large the number is.
For '0' it is ~30% regressions.

Reply:
> The alternative is to do this as runtime check against pointer size

Is there a safe way to do this that doesn't require the Unsafe class? Or is it OK to use Unsafe here even if we remove the other uses of Unsafe within the implementations?

Reply:
Can I just do the following instead?
```C#
if (IntPtr.Size == 8)
```

Why do we use Unsafe, for instance here: https://github.com/dotnet/corefx/blob/master/src/System.Memory/tests/Span/Overflow.cs#L24?

Reply:
All `IntPtr.Size`, `Unsafe.SizeOf<IntPtr>()` and `sizeof(IntPtr)` are pretty similar. Depends on your personal preferences which one to use.

`IntPtr.Size` and `Unsafe.SizeOf<IntPtr>()` will turn into method calls that the JIT have to inline to figure out what they are just constants. `sizeof(IntPtr)` is JIT time constant, so it allows the JIT to generate better code - especially in your case with a ton of dead force-inlined methods.

There is nothing really unsafe about `sizeof(IntPtr)` - it is just C# being overprotective.

Reply:
There are quite a few uses of IntPtr.Size in corefx, sounds like they could safely be replaced with sizeof(IntPtr).

Reply:
> There are quite a few uses of IntPtr.Size in corefx, sounds like they could safely be replaced with sizeof(IntPtr).

Except, we would have to then wrap all those calls around an `unsafe` block.